### PR TITLE
autoClose Contribution

### PR DESCRIPTION
**Pull request by MrRedBeard**
There wasn't a branch for this.

Added option for autoClose on html but set to false.

Added polyfills for hideEl and showEl at end of js file.

Added logic to hide claendar when date is set for both single date and date range on line 153 of js.

A placeholder for the element needs to be added so that the calendar can be accessed after the calendar is closed.

### CODE REVIEW

File: Datepicker.js [hunk 1]
```diff
@@ -147,6 +147,22 @@ class clsDatepicker {
             this.containerElement.querySelector('.startDateElement').innerHTML = "Date: " + this.dates[0];
         }
         this.highlightDates();
+		
+		//autoClose the claendar when a single date or date range is selected
```
Comment (ctrachte) [inline Datepicker.js:151]: "claendar" -  correct spelling (or just remove the comment)

File: Datepicker.js [hunk 2]
```diff
@@ -147,6 +147,22 @@ class clsDatepicker {
...
+				this.containerElement.hideEl();
+			}.bind(this), 400);
+		}
+		else if(this.singleDate && this.dates.length === 1 && this.options.autoClose)
+		{
+			setTimeout(function()
+			{ 
+				this.containerElement.hideEl();
+			}.bind(this), 400);
+		}
```
Comment (ctrachte) [inline Datepicker.js:165]: back up indentation to match the rest of the code formatting

**Review by ctrachte — CHANGES_REQUESTED** [review]
Looks good so far! Only minor formatting fix requested.

File: Datepicker.js (hunk 2 above)
Comment (ctrachte) [reply Datepicker.js:165]: If your version of VSCode won't format this properly I can still merge and fix it later on the master branch.

Comment (MrRedBeard) [reply Datepicker.js:165]: Not even Github's editor formats like this. You know my feelings about it but this is your project.

**Comment by MrRedBeard** [comment]
Do I have to do an additional pull request?

**Review by ctrachte — APPROVED** [review]
Perfect! Thanks!

**Pull request merged**